pkgs/development/python-modules/rauth/default.nix

```diff
@@ -0,0 +1,38 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, requests
+, flake8
+}:
+
+buildPythonPackage rec {
+  pname = "rauth";
+  version = "0.7.2";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "litl";
+    repo = "rauth";
+    rev = version;
+    hash = "sha256-wRKZbxZCEfihOaJM8sk8438LE++KJWxdOGImpL1gHa4=";
+  };
+
+  propagatedBuildInputs = [
+    requests
+  ];
+
+  nativeCheckInputs = [
+    flake8
+  ];
```

Comment:
`flake8` is a linter, and its success or failure does not affect the behavior of the package.
So we do not need to check it again here.

Reply:
We can use `unittestCheckHook` here or we may need to define our own checkPhase.

Reply:
@natsukium what do you thing of the latest force-push? I use the pypi release which doesn't apparently include any tests:
```
python3.10-rauth> make: *** No rule to make target 'check'.  Stop.
error: builder for '/nix/store/7ljybj0b9fz8nj5vy8a93zp9n7j4wims-python3.10-rauth-0.7.3.drv' failed with exit code 2
```

Reply:
In such cases, we can use something like `fetchFromGitHub` to fetch the source directly from the VCS.
However, since this package is related to networking, it may not pass most tests.

Reply:
Thanks to your various guidance on the various PRs, I ended up finding the solution with proper texting. :smile: